rules/TypeDeclaration/Rector/ClassMethod/NarrowObjectReturnTypeRector.php

```diff
@@ -241,56 +235,6 @@ private function isNarrowingValid(string $declaredType, string $actualType): boo
             ->yes();
     }
 
-    private function hasParentMethodWithNonObjectReturn(ClassMethod $classMethod): bool
```

Comment:
The `isNarrowingValid` logic can be added on verify parent method return

Reply:
Can we finish this one up and merge it?

Reply:
hey sorry i was very busy, i will do this today